device/alpaka/src/utils/magnetic_field_types.hpp

```diff
@@ -8,12 +8,50 @@
 #pragma once
 
 #include "traccc/bfield/magnetic_field_types.hpp"
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+#include <covfie/cuda/backend/primitive/cuda_device_array.hpp>
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+#include <covfie/hip/backend/primitive/hip_device_array.hpp>
+#elif defined(ALPAKA_ACC_SYCL_ENABLED)
+#include <covfie/sycl/backend/primitive/sycl_device_array.hpp>
+#endif
 
 namespace traccc::alpaka {
 
+/// Inhomogeneous B-field backend type for Alpaka
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+/// Inhomogeneous B-field backend type using CUDA global memory
```

Comment:
This is all copied from the CUDA and SYCL magnetic_field_types: if these were public headers then could avoid some code there. This would also mean that we could move the HIP magnetic field code to `device/hip`.